src/main/java/uk/gov/laa/ccms/caab/client/EbsApiClient.java

```diff
@@ -285,6 +306,76 @@ public Mono<ProceedingDetail> getProceeding(final String proceedingCode) {
         .onErrorResume(e -> ebsApiClientErrorHandler.handleProceedingError(proceedingCode, e));
   }
 
+  /**
+   * Retrieves proceeding details.
+   *
+   * @return A Mono containing the ProceedingDetails or an error handler if an error occurs.
+   */
+  public Mono<ProceedingDetails> getProceedings(
+      final String categoryOfLaw,
+      final String matterType,
+      final Boolean isAmendment,
+      final Boolean larScopeFlag,
+      final String applicationType,
+      final boolean isLead) {
+    return ebsApiWebClient
+        .get()
+        .uri(builder -> builder.path("/proceedings")
+            .queryParamIfPresent("category-of-law", Optional.ofNullable(categoryOfLaw))
+            .queryParamIfPresent("matter-type", Optional.ofNullable(matterType))
+            .queryParamIfPresent("amendment-only", Optional.ofNullable(isAmendment))
+            .queryParamIfPresent("lar-scope-flag", Optional.ofNullable(larScopeFlag))
+            .queryParamIfPresent("application-type", Optional.ofNullable(applicationType))
+            .queryParamIfPresent("lead", Optional.of(isLead))
```

Comment:
If it's primitive, it has to be present?

Reply:
amended